nengo_loihi/loihi_cx.py

```diff
@@ -101,17 +102,25 @@ def add_probe(self, probe):
         assert probe.target is self
         self.probes.append(probe)
 
-    def set_decay_U(self, tau_s, dt):
-        self.decayU[:] = 1 if tau_s == 0 else -np.expm1(-dt/np.asarray(tau_s))
+    def set_decay_U(self, tau_s, dt, default=False, overwrite=False):
+        if default and self.decayU_set:
+            return
 
-    def configure_filter(self, tau_s, dt=0.001):
-        """Synaptic input filter for Cx."""
+        decayU = 1 if tau_s == 0 else -np.expm1(-dt/np.asarray(tau_s))
+        if not overwrite and self.decayU_set and not np.allclose(
+                decayU, self.decayU):
+            raise ValueError(
+                "Cannot change tau_s on already configured neurons")
+
+        self.decayU[:] = decayU
+        self.decayU_set = not default
 
-        self.set_decay_U(tau_s, dt)
+    def configure_filter(self, tau_s, dt=0.001, **kwargs):
+        """Synaptic input filter for Cx."""
+        self.set_decay_U(tau_s, dt, **kwargs)
 
-    def configure_lif(
-            self, tau_s=0.005, tau_rc=0.02, tau_ref=0.001, vth=1, dt=0.001):
-        self.set_decay_U(tau_s, dt)
+    def configure_lif(self, tau_rc=0.02, tau_ref=0.001, vth=1, dt=0.001):
+        self.set_decay_U(0.005, dt, default=True)
```

Comment:
What does `tau_s` represent? I don't think it's documented anywhere but documenting it would help in determining whether it makes sense to hard code it for LIF and ReLu.

Reply:
It's the synapse time constant on the neuron input. It's not hard-coded here (`default=True`), but I agree, it should be documented. The default actually probably doesn't have to be set here at all, and in fact maybe it shouldn't.